### PR TITLE
feat: add storage metrics for container collector

### PR DESCRIPTION
**Pull request by szediktam**
Add storage metrics for windows container from container's storagestats of hcsshim.
reference: [windows container's Statistics struct](<https://github.com/microsoft/hcsshim/blob/master/test/vendor/github.com/Microsoft/hcsshim/internal/hcs/schema1/schema1.go#L188>)

Example metrics:
```
# TYPE windows_container_storage_read_count_normalized_total counter
windows_container_storage_read_count_normalized_total{container_id="docker://0a1d78f90178bbfd80aa49572a60868c2421899121c60394938ca2e830860a7c"} 5088
windows_container_storage_read_count_normalized_total{container_id="docker://1a4e8d1fdbe62dfa8638cf90704f72d9a199e3afccfc1b36d364a96fb72577ad"} 1213
windows_container_storage_read_count_normalized_total{container_id="docker://38c1eb0b679a9dcfe2ab259617b653740241caaf8b6018770b2977dcb8762ec3"} 897
windows_container_storage_read_count_normalized_total{container_id="docker://be0a3b99bbd220a0fe30db27cbb11899485815d776d687e3e68afcb38890c6ee"} 942
windows_container_storage_read_count_normalized_total{container_id="docker://cd83f591f99d6530424632ea4a5cbbaf44f1e51eb8d4683d020480ab21be451a"} 5038
windows_container_storage_read_count_normalized_total{container_id="docker://eccf6290641c9e2bbe9d71ad8d1210498584f70a520ebe78f6812708bf406018"} 1109
# HELP windows_container_storage_read_size_bytes_total Read Size Bytes
# TYPE windows_container_storage_read_size_bytes_total counter
windows_container_storage_read_size_bytes_total{container_id="docker://0a1d78f90178bbfd80aa49572a60868c2421899121c60394938ca2e830860a7c"} 3.8066176e+07
windows_container_storage_read_size_bytes_total{container_id="docker://1a4e8d1fdbe62dfa8638cf90704f72d9a199e3afccfc1b36d364a96fb72577ad"} 9.13408e+06
windows_container_storage_read_size_bytes_total{container_id="docker://38c1eb0b679a9dcfe2ab259617b653740241caaf8b6018770b2977dcb8762ec3"} 6.810112e+06
windows_container_storage_read_size_bytes_total{container_id="docker://be0a3b99bbd220a0fe30db27cbb11899485815d776d687e3e68afcb38890c6ee"} 7.121408e+06
windows_container_storage_read_size_bytes_total{container_id="docker://cd83f591f99d6530424632ea4a5cbbaf44f1e51eb8d4683d020480ab21be451a"} 3.7699584e+07
windows_container_storage_read_size_bytes_total{container_id="docker://eccf6290641c9e2bbe9d71ad8d1210498584f70a520ebe78f6812708bf406018"} 8.397312e+06
# HELP windows_container_storage_write_count_normalized_total Write Count Normalized
# TYPE windows_container_storage_write_count_normalized_total counter
windows_container_storage_write_count_normalized_total{container_id="docker://0a1d78f90178bbfd80aa49572a60868c2421899121c60394938ca2e830860a7c"} 37673
windows_container_storage_write_count_normalized_total{container_id="docker://1a4e8d1fdbe62dfa8638cf90704f72d9a199e3afccfc1b36d364a96fb72577ad"} 841
windows_container_storage_write_count_normalized_total{container_id="docker://38c1eb0b679a9dcfe2ab259617b653740241caaf8b6018770b2977dcb8762ec3"} 837
windows_container_storage_write_count_normalized_total{container_id="docker://be0a3b99bbd220a0fe30db27cbb11899485815d776d687e3e68afcb38890c6ee"} 863
windows_container_storage_write_count_normalized_total{container_id="docker://cd83f591f99d6530424632ea4a5cbbaf44f1e51eb8d4683d020480ab21be451a"} 37617
windows_container_storage_write_count_normalized_total{container_id="docker://eccf6290641c9e2bbe9d71ad8d1210498584f70a520ebe78f6812708bf406018"} 881
# HELP windows_container_storage_write_size_bytes_total Write Size Bytes
# TYPE windows_container_storage_write_size_bytes_total counter
windows_container_storage_write_size_bytes_total{container_id="docker://0a1d78f90178bbfd80aa49572a60868c2421899121c60394938ca2e830860a7c"} 2.26397696e+08
windows_container_storage_write_size_bytes_total{container_id="docker://1a4e8d1fdbe62dfa8638cf90704f72d9a199e3afccfc1b36d364a96fb72577ad"} 6.148096e+06
windows_container_storage_write_size_bytes_total{container_id="docker://38c1eb0b679a9dcfe2ab259617b653740241caaf8b6018770b2977dcb8762ec3"} 6.090752e+06
windows_container_storage_write_size_bytes_total{container_id="docker://be0a3b99bbd220a0fe30db27cbb11899485815d776d687e3e68afcb38890c6ee"} 6.352896e+06
windows_container_storage_write_size_bytes_total{container_id="docker://cd83f591f99d6530424632ea4a5cbbaf44f1e51eb8d4683d020480ab21be451a"} 2.26188288e+08
windows_container_storage_write_size_bytes_total{container_id="docker://eccf6290641c9e2bbe9d71ad8d1210498584f70a520ebe78f6812708bf406018"} 6.377472e+0
```

### CODE REVIEW

**Comment by szediktam** [comment]
@carlpett @sachinmsft Hi, I add some storage metrics for windows container. Please reveiw it. Thanks.

**Review by breed808 — COMMENTED** [review]
Thanks for submitting this! It looks good to me, aside from one minor issue with the metric names.

I'm not able to test this myself; have you been able to test the change to confirm the new metrics are being exposed by the collector?

File: collector/container.go [hunk 1]
```diff
@@ -127,6 +133,30 @@ func NewContainerMetricsCollector() (Collector, error) {
...
+			"Write Count Normalized",
+			[]string{"container_id"},
+			nil,
+		),
+		WriteSizeBytes: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "storage_write_size_bytes"),
+			"Write Size Bytes",
+			[]string{"container_id"},
+			nil,
+		),
```
[the remark this answers is not in the document]
Comment (szediktam) [reply collector/container.go]: Oh, yes.  `_total` suffix is a convention for counters. Thanks.
I update the pr, please review again. I also update the example metrics in the first comment.

**Comment by szediktam** [comment]
> Thanks for submitting this! It looks good to me, aside from one minor issue with the metric names.
> 
> I'm not able to test this myself; have you been able to test the change to confirm the new metrics are being exposed by the collector?

Thanks for your review. I update the pr and update the example metrics in the first comment. Please review again.

**Comment by szediktam** [comment]
@breed808  This PR also add a useful queries example for container docs. Please review again. Thanks.

**Review by breed808 — APPROVED** [review]
@szediktam I've re-reviewed and all looks good :+1:

Note that I've had to rebase this PR on `master` to fix the CI jobs.

**Pull request merged**